### PR TITLE
AA-1262: Disable times before now and before minDate and after maxDate

### PR DESCRIPTION
**Pull request by coughlanio**

### CODE REVIEW

File: pikaday.js [hunk 1]
```diff
@@ -470,18 +468,17 @@
...
 
-                    isJustOnTime = date.valueOf() === self._minTime.valueOf();
-                    if (isJustOnTime) {
-                        disabled = false;
-                    } else {
-                        disabled = (isMinDate && date <= self._minTime) || (isMaxDate && date >= self._maxTime);
-                    }
+                    disabled = false;
+
+                    if (date.getTime() < self._minTime.getTime()) disabled = true;
```
Comment (briandeheus) [inline pikaday.js]: If you want to write conditions without blocks:

```
disabled = ((date.getTime() < self._minTime.getTime()) || (date.getTime() > self._maxTime.getTime()))
```

Comment (coughlanio) [reply pikaday.js]: I originally had something similar, and to me it's not as readable, but if the majority feels like it's better the other way, then I can action the change 😄

Comment (briandeheus) [reply pikaday.js]: It's not about readability as much as it's about:

https://stackoverflow.com/a/42416541/2465738

**Comment by hachibeeDI** [comment]
After write the HTML and JavaScript code in `index.html` like as 

```html
    <input id="datepicker" type="text">
```

```javascript
    var picker = new Pikaday(
    {
        field: document.getElementById('datepicker'),
        firstDay: 1,
        minDate: new Date(2017, 5, 10, 12, 20),
        maxDate: new Date(2017, 11, 15, 17, 20),
        showTime: true,
    });
```

and open it by browser, minDate and maxDate is one month later. 🤔

**Comment by hachibeeDI** [comment]
Ahh, I've forgotten about the silly `Date`'s interface.  It works correct.  Month is zero start. :sadpanda:

**Comment by rmcauley** [comment]
:horse:

**Comment by hachibeeDI** [comment]
🐴

**Pull request merged**